### PR TITLE
Feature changes

### PR DESCRIPTION
**Pull request by Incompleteusern**
This pr rewrites `Feature` to be more general by removing the `apis` and `keybinds` list, instead letting subclasses load them if necessary. Furthermore, it shifts `WebManager` to statically loads apis as well.

This is currently a draft due to depending on #13 as well as being as of now untested

### CODE REVIEW

File: common/src/main/java/com/wynntils/core/features/Feature.java [hunk 1]
```diff
@@ -48,10 +39,15 @@ public Feature() {
         }
     }
 
-    protected void init(
-            ImmutableList.Builder<WebProviderSupplier> apis,
-            ImmutableList.Builder<KeySupplier> keybinds,
-            ImmutableList.Builder<Condition> conditions) {
+    protected void init(ImmutableList.Builder<Condition> conditions) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/features/Feature.java]: You can make a method abstract to force subclasses to implement it...

File: common/src/main/java/com/wynntils/core/features/Feature.java [hunk 2]
```diff
@@ -60,48 +56,14 @@ protected void init(
      *
      * <p>Returns whether the feature was successfully activated
      */
-    public final boolean enable() {
+    public final void enable() {
         if (enabled)
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/features/Feature.java:55]: I realize this is not new code, but I really really dislike the use of multiline if statements without braces. I'd like to vote for having a formal rule against it; preferably checked by style checkers if possible...

File: common/src/main/java/com/wynntils/core/webapi/WebManager.java [hunk 3]
```diff
@@ -28,45 +26,37 @@
...
-                                .build();
-                    }
-                };
+    private static void loadItemGuesses() {
+        handler.addRequest(
+                new RequestBuilder(apiUrls.get("ItemGuesses"), "item_guesses")
+                        .cacheTo(new File(API_CACHE_ROOT, "item_guesses.json"))
+                        .handleJsonObject(
+                                json -> {
+                                    Type type =
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/webapi/WebManager.java]: spotless is really having a weird opinion on how good code should look. Maybe we should go back to the drawing board and start looking for another formatter. This looks horrible and almost unreadable, imho.

**Review by magicus — COMMENTED** [review]
Overall it seems like a reasonable refactoring. Bit hard to tell due to #13 mixed in with this.

File: common/src/main/java/com/wynntils/core/features/Feature.java (hunk 1 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/features/Feature.java]: I want to make it optional to extend

File: common/src/main/java/com/wynntils/core/features/Feature.java (hunk 2 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/features/Feature.java:55]: I’m fine either way, get a consensus and I’ll follow the guideline

File: common/src/main/java/com/wynntils/core/webapi/WebManager.java (hunk 3 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/core/webapi/WebManager.java]: Yeah, lambdas/builders aren’t handled well

**Comment by Incompleteusern** [comment]
Yeah the only real changes are in [082da0](https://github.com/Wynntils/Artemis/pull/14/commits/082da0df7df12c3128adc6748b38f911fbdd9783)

**Comment by Incompleteusern** [comment]
Nvm, the new commit of [7e0166](https://github.com/Wynntils/Artemis/pull/14/commits/7e0166fb53b7446204db17bb3231bc9d25943fa0) also changes stuff

**Review by kristofbolyai — APPROVED** [review]
Looks good to me!

**Review by magicus — APPROVED** [review]
I'm getting a bit lost in all changes, but I think this looks good. Anyway, push it, and if it needs to be polished, we can do that later.

In general, I think this is a sane way of developing -- as we notice that more structured APIs are needed, we refactor the code base to get there. It's hard to get a proper architecture correct in advance of actually using it.

**Pull request merged**